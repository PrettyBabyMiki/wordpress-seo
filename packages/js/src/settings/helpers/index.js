export * from "./i18n";
export * from "./user-social-profiles";
export * from "./search";
export * from "./submit";
export * from "./validation";
