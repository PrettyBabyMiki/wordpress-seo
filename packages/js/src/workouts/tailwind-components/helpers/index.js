import classNames from "classnames";

/**
 * Creates the error ID for the error component.
 *
 * @param {string} inputId The id of the input component.
 *
 * @returns {string} The ID for the error component.
 */
export const getErrorId = ( inputId ) => `error-${ inputId }`;

/**
  * Get props needed to properly display an error in input components.
  *
  * @param {string} inputId The id of the input component.
  * @param {ValidationError} error The error object.
  * @param {boolean} error.isVisible The error object.
  *
  * @returns {Object} Object containing relevant props for displaying.
  */
export const getErrorAriaProps = ( inputId, { isVisible } ) => isVisible ? {
	"aria-invalid": true,
	"aria-describedby": getErrorId( inputId ),
} : {};

/**
 * Helper function to get active styles for select options.
 *
 * @param {boolean} options.active Whether the option is active.
 *
 * @returns {string} Styles for an active option.
 */
export function getOptionActiveStyles( { active, selected } ) {
	return classNames(
		"yst-relative yst-cursor-default yst-select-none yst-py-2 yst-pl-3 yst-pr-9 yst-my-0",
		selected && "yst-bg-primary-500 yst-text-white",
		( active && ! selected ) && "yst-bg-primary-200 yst-text-gray-900",
		( ! active && ! selected ) && "yst-text-gray-900"
	);
}
