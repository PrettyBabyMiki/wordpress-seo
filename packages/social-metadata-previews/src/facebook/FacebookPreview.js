/* External dependencies */
import React, {Component} from "react";
import PropTypes from "prop-types";
import styled from "styled-components";

/* Internal dependencies */
import FacebookSiteAndAuthorNames from "./FacebookSiteAndAuthorNames";
import FacebookImage from "./FacebookImage";
import FacebookTitle from "./FacebookTitle";
import FacebookDescription from "./FacebookDescription";

const FacebookPreviewWrapper = styled.div`
	display: flex;
	flex-wrap: wrap;
	background-color: #F2F3F5;
	max-width: none;
	border: none;
	box-shadow: none;
	border: 1px solid #dddfe2;
	border-radius: 0;
	zoom: 1;
	overflow: hidden;
	position: relative;
	z-index: 0;

	width:500px;
`;

const OuterTextWrapper = styled.div`
	background-color: #f2f3f5;
    border-bottom: 1px solid #dddfe2;
    margin: 0;
    padding: 10px 12px;
	position: relative;
	
	width: 500px;
	height: 57px;
`;

const InnerTextWrapper = styled.div`
	max-height: 190px;
`;

const TitleAndDescriptionWrapper = styled.div`
`;

/**
 * Renders FacebookPreview component.
 *
 * @param {object} props The props.
 *
 * @returns {React.Element} The rendered element.
 */
class FacebookPreview extends Component {
	/**
	 * The constructor.
	 *
	 * @param {Object} props The component's props.
	 */
	constructor( props ) {
		super( props );
		this.state = {
			imageType:''
		};
	}

	getImageMode (imageMode) {
		this.setState({ imageType: imageMode })
	}

	render() {
		return (
			<FacebookPreviewWrapper>
				<FacebookImage src={ this.props.image } alt={ this.props.alt } getImageMode={() => this.getImageMode}  />
					<OuterTextWrapper>
						<InnerTextWrapper>
							<FacebookSiteAndAuthorNames siteName={ this.props.siteName } authorName={ this.props.authorName } />
								<TitleAndDescriptionWrapper>
									<FacebookTitle title={ this.props.title } />
									<FacebookDescription>
										{ this.props.description }
									</FacebookDescription>
								</TitleAndDescriptionWrapper>
						</InnerTextWrapper>
					</OuterTextWrapper>
			</FacebookPreviewWrapper>
		);
};

}
FacebookPreview.propTypes = {
	siteName: PropTypes.string.isRequired,
	title: PropTypes.string.isRequired,
	authorName: PropTypes.string,
	description: PropTypes.string,
	image: PropTypes.string.isRequired,
	alt: PropTypes.string,
};

FacebookPreview.defaultProps = {
	authorName: "",
	description: "",
	alt: "",
};


export default FacebookPreview;
