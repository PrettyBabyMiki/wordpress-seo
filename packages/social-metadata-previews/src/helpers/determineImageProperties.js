export const TWITTER_IMAGE_SIZES = {
	"squareWidth": 123,
	"squareHeight": 123,
	"landscapeWidth": 506,
	"landscapeHeight": 253
};

export const FACEBOOK_IMAGE_SIZES = {
	"squareWidth": 158,
	"squareHeight": 158,
	"landscapeWidth": 500,
	"landscapeHeight": 261,
	"portraitWidth": 158,
	"portraitHeight": 236
};

/**
 * Determines the image display mode.
 *
 * @param {string} socialMedium Facebook or Twitter.
 * @param {Object} dimensions The image's dimensions.
 *
 * @returns {string} The display mode of the image.
 */
export function determineImageMode( socialMedium, dimensions ) {
	if ( socialMedium === "Facebook" ) {
		if ( dimensions.height > dimensions.width ) {
			return "portrait";
		}

		if ( dimensions.height === dimensions.width ) {
			return "square";
		}

		return "landscape";
	}

	if ( socialMedium === "Twitter" ) {
		if ( dimensions.height > 157 && dimensions.width > 300 ) {
			return "landscape";
		}

		return "square";
	}
}

/**
 * Retrieves the image sizes.
 *
 * @param {string} socialMedium Facebook or Twitter.
 *
 * @returns {Object} Object containing the image sizes.
 */
export function retrieveImageSizes ( socialMedium ) {
	if ( socialMedium === "Twitter" ) {
		return TWITTER_IMAGE_SIZES;
	}

	return FACEBOOK_IMAGE_SIZES;
}

/**
 * Gets the dimensions of the uploaded image.
 *
 * @param {string} src The image source.
 *
 * @returns {Object} The original image dimensions.
 */
export function retrieveOriginalImageDimensions( src ) {
	/* istanbul ignore next */
	return new Promise( ( resolve, reject ) => {
		const img = new Image();

		img.onload = () => {
			resolve( {
				width: img.width,
				height: img.height,
			} );
		};

		img.onerror = reject;

		img.src = src;
	} );
}

/**
 * Calculates the ratios of the width and height of the original image in relation to the width and
 * height of the expected image.
 *
 * When we're in landscape mode, we can't just resize to the dimensions expected by Twitter.
 * If we'd do so, we would end up with warped images. That's why we calculate the ratio between
 * the original width and height and the width and height that is expected by Twitter.
 * For example: the original image is 898x1600 (height x width) and Twitter expects 253x506.
 * The heightRatio would be 3.55 (898/253) and the widthRatio would be 3.16 (1600/506).
 *
 * @param {Object} imageSizes The sizes of images in the social medium.
 * @param {Object} dimensions The dimensions of the original image.
 * @param {string} imageMode The image mode: square or landscape.
 *
 * @returns {Object} The image's width ratio and height ratio.}
 */
export function calculateImageRatios( imageSizes, dimensions, imageMode ) {

	if ( imageMode === "landscape" ) {
		return {
			widthRatio: dimensions.width / imageSizes.landscapeWidth,
			heightRatio: dimensions.height / imageSizes.landscapeHeight,
		};
	}

	if ( imageMode === "portrait" ) {
		return {
			widthRatio: dimensions.width / imageSizes.portraitWidth,
			heightRatio: dimensions.height / imageSizes.portraitHeight,
		};
	}

	if ( imageMode === "square" ) {
		return {
			widthRatio: dimensions.width / imageSizes.squareWidth,
			heightRatio: dimensions.height / imageSizes.squareHeight,
		};
	}
}

/**
 * Calculates the largest dimensions that can be used for rendering the image.
 *
 * To use as much as the allowed space as possible, we base both dimensions on the dimension with
 * the lowest imageRatio (see above). For example: for a 898x1600 image, the heightRatio is larger
 * than the widthRatio. The result of dividing by the widthRatio is a 284x506 image.
 * The excess of 284-253 = 31 pixels will be 'cut off' by the container in the presentation part.
 *
 * If we would divide by the heightRatio, the image would become 253x451, which means it would not
 * be wide enough for the container, which means there would be a 506-451=55px white border on one
 * of the sides.
 *
 * @param {Object} dimensions  The dimensions of the original image.
 * @param {Object} imageRatios The ratios of the width and height of the original image in relation
 *                             to the width and height of the expected image.
 *
 * @returns {Object}           The width and height that the image should have as Twitter/Facebook
 *                             image.
 */
export function calculateLargestDimensions( dimensions, imageRatios ) {
	if ( imageRatios.widthRatio <= imageRatios.heightRatio ) {
		return {
			width: dimensions.width / imageRatios.widthRatio,
			height: dimensions.height / imageRatios.widthRatio,
		};
	}

	return {
		width: dimensions.width / imageRatios.heightRatio,
		height: dimensions.height / imageRatios.heightRatio,
	};
}

/**
 * Calculates the dimensions of the image to use as Twitter/Facebook image.
 *
 * @param {Object} imageSizes The sizes of images in the social medium.
 * @param {Object} originalDimensions The width and height of the original image.
 * @param {string} imageMode The image mode: square or landscape.
 *
 * @returns {object} The image dimensions.
 */
export function calculateImageDimensions( imageSizes, originalDimensions, imageMode ) {
	// Images that are too small should not be scaled.
	if ( originalDimensions.width < imageSizes.squareWidth ||
		 originalDimensions.height < imageSizes.squareHeight ) {
		return {
			width: originalDimensions.width,
			height: originalDimensions.height,
		};
	}

	/*
	 * If the image should be rendered as a square, and its original dimensions were also square,
	 * just use the squareWidth and squareHeight. We don't have to fear that the resulting image
	 * will be warped.
	 */
	if ( imageMode === "square" ) {
		if ( originalDimensions.width === originalDimensions.height ) {
			return {
				width: imageSizes.squareWidth,
				height: imageSizes.squareHeight,
			};
		}

		/*
		 * If the image should be rendered as a square, but originally wasn't square, crop the
		 * longest side. This way, the image won't be warped.
		 */
		const imageRatios = calculateImageRatios( imageSizes, originalDimensions, imageMode );

		return calculateLargestDimensions( originalDimensions, imageRatios );
	}

	/*
	 * If the image should be rendered as a landscape or portrait, crop the longest side, to reach
	 * the required size ratio. This way, the image won't be warped.
	 */
	if ( imageMode === "landscape" || imageMode === "portrait") {
		const imageRatios = calculateImageRatios( imageSizes, originalDimensions, imageMode );

		return calculateLargestDimensions( originalDimensions, imageRatios );
	}
}

/**
 * Determines the properties of the image.
 *
 * @param {string} src The source of the image.
 * @param {string} socialMedium Facebook or Twitter.
 *
 * @returns {Promise} The promise of the imageProperties.
 */
export function determineImageProperties( src, socialMedium ) {
	return retrieveOriginalImageDimensions( src ).then( ( originalDimensions ) => {
		// Determine what image mode should be used.
		const imageMode = determineImageMode( socialMedium, originalDimensions );

		// Retrieve the image sizes, depending on the social medium.
		const imageSizes = retrieveImageSizes( socialMedium );

		// Calculate the image dimensions for the specific image.
		const ImageDimensions = calculateImageDimensions( imageSizes, originalDimensions, imageMode );

		return {
			mode: imageMode,
			height: ImageDimensions.height,
			width: ImageDimensions.width,
		};
	} );
}
