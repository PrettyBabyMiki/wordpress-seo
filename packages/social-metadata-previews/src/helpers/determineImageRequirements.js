/** Checks the if the filesize exceeds 5MB. If not correct returns a warning.
 *
 * @param {String} imageBytes The filesize in MB.
 * @returns {String | Void} A warning string.
 */
const checkImageBytes = ( imageBytes ) => {
	if ( imageBytes > 5 ) {
		return "The file size of the uploaded image is too large for Twitter. File size must be less than 5MB.";
	}
	return;
};

/** Checks if the image type is allowed. If not correct returns a warning.
 *
 * @param {String} type The image type.
 * @returns {String | Void} A warning string.
 */
const checkImageType = ( type ) => {
	switch ( type ) {
		case "JPG":
		case "JPEG":
		case "PNG":
		case "WEBP":
			break;
		case "GIF":
			return "You have uploaded a GIF. Please note that, if itâ€™s an animated GIF, only the first frame will be used.";
		default:
			return "The format of the uploaded image is not supported. The supported formats are: JPG, PNG, WEBP and GIF.";
	}
};

/** Checks if the height and the width of an image are correct. If not correct returns a warning.
 *
 * @param {Number} height The image height.
 * @param {Number} width The image width.
 *
 * @returns {String | Void} A warning string.
 */
const checkTwitterImageSize = ( height, width ) => {
	const warningMessage = "Your image dimensions are not suitable: " +
	"The minimum dimensions are 200x200 pixels. The maximum dimensions are 4096x4096 pixels.";

	if ( height < 200 || width < 200 ) {
		return warningMessage;
	}
	if ( height > 4096 || width > 4096 ) {
		return warningMessage;
	}
	return;
};

/** Checks if the height and the width of an image are correct. If not correct returns a warning.
 *
 * @param {Number} height The image height.
 * @param {Number} width The image width.
 *
 * @returns {String | Void} A warning string.
 */
const checkFacebookImageSize = ( height, width ) => {
	if ( height < 200 || width < 200 ) {
		return "The uploaded image is too small. The minimum dimensions for this image are 200x200 pixels.";
	}
	return;
};

/**
 * @param {Object} image The image Object.
 *
 * @returns {String[]} An Array containing warnings.
 */
export const checkTwitterRequirements = ( image ) => {
	const { bytes, height, width, type } = image;

	let warningsArray = [];

	const imageBytesResult = checkImageBytes( bytes );
	const imageTypeResult = checkImageType( type );
	const imageSizeResult = checkTwitterImageSize( height, width );

	if ( imageBytesResult ) {
		warningsArray = [ ...warningsArray, imageBytesResult ];
	}
	if ( imageTypeResult ) {
		warningsArray = [ ...warningsArray, imageTypeResult ];
	}
	if ( imageSizeResult ) {
		warningsArray = [ ...warningsArray, imageSizeResult ];
	}
	return warningsArray;
};

/**
 * @param {Object} image The image Object.
 * @returns {String[]} An Array containting warning.
 */
export const checkFacebookRequirements = ( image ) => {
	const { height, width } = image;

	let warningsArray = [];

	const imageSizeResult = checkFacebookImageSize( height, width );
	if ( imageSizeResult ) {
		warningsArray = [ ...warningsArray, imageSizeResult ];
	}

	return warningsArray;
};
