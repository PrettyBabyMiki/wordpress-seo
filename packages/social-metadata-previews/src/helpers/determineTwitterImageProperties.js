export const SQUARE_WIDTH = 123;
export const SQUARE_HEIGHT = 123;
export const LANDSCAPE_WIDTH = 506;
export const LANDSCAPE_HEIGHT = 253;

/**
 * Determines the image display mode of the Twitter image.
 *
 * @param {Object} cardType The card type in which an image should be displayed.
 *
 * @returns {string} The display mode of the image.
 */
export function determineTwitterImageMode( cardType ) {
	if ( cardType === "summary" ) {
		return "square";
	}

	return "landscape";
}

/**
 * Gets the dimensions of the uploaded image.
 *
 * @param {string} src The image source.
 *
 * @returns {Object} The image original image dimensions.
 */
export function getOriginalImageDimensions( src ) {
	/* istanbul ignore next */
	return new Promise( ( resolve, reject ) => {
		const img = new Image();

		img.onload = () => {
			resolve( {
				width: img.width,
				height: img.height,
			} );
		};

		img.onerror = reject;

		img.src = src;
	} );
}

/**
 * Gets the ratios of the width and height of the original image in relation to the width and height
 * of the expected image.
 *
 * When we're landscape mode, we can't just resize to the dimensions expected by Twitter. If we'd do
 * so, we would end up with warped images. That's why we calculate the ratio between the original width and height and
 * the width and height that is expected by Twitter. For example: the original image is 898x1600 (height x width) and
 * Twitter expects 253x506. The heightRatio would be 3.55 (898/253) and the widthRatio would be 3.16 (1600/506).
 *
 * @param {Object} dimensions The dimensions of the original image.
 * @param {string} imageMode The image mode: square or landscape.
 *
 * @returns {Object} The image's width ratio and height ratio.}
 */
export function getImageRatios( dimensions, imageMode ) {
	if ( imageMode === "landscape" ) {
		return {
			widthRatio: dimensions.width / LANDSCAPE_WIDTH,
			heightRatio: dimensions.height / LANDSCAPE_HEIGHT,
		};
	}

	if ( imageMode === "square" ) {
		return {
			widthRatio: dimensions.width / SQUARE_WIDTH,
			heightRatio: dimensions.height / SQUARE_HEIGHT,
		};
	}
}

/**
 * Gets the image dimensions that the image should have as Twitter image.
 *
 * To use as much as the allowed space as possible, we base both dimensions on the dimension with the lowest imageRatio
 * (see above). For example: for a 898x1600 image, the heightRatio is larger than the widthRatio. The result of dividing
 * by the widthRatio is a 284x506 image. The excess of 284-253 = 31 pixels will be 'cut off' by the
 * container in the presentation part.
 *
 * If we would divide by the heightRatio, the image would become 253x451, which means it would not be wide enough for
 * the container, which means there would be a 506-451=55px white border on one of the sides.
 *
 * @param {Object} dimensions  The dimensions of the original image.
 * @param {Object} imageRatios The ratios of the width and height of the original image in relation to the width and
 *                             height of the expected image.
 *
 * @returns {Object}           The width and height that the image should have as Twitter image.
 */
export function getImageDimensionsForTwitterImage( dimensions, imageRatios ) {
	if ( imageRatios.widthRatio <= imageRatios.heightRatio ) {
		return {
			width: dimensions.width / imageRatios.widthRatio,
			height: dimensions.height / imageRatios.widthRatio,
		};
	}

	return {
		width: dimensions.width / imageRatios.heightRatio,
		height: dimensions.height / imageRatios.heightRatio,
	};
}

/**
 * Calculates the dimensions of the image to use as TwitterImage.
 *
 * @param {Object} originalDimensions The width and height of the original image.
 * @param {string} imageMode The image mode: square or landscape.
 *
 * @returns {object} The image dimensions for the Twitter image.
 */
export function calculateTwitterImageDimensions( originalDimensions, imageMode ) {
	// Images that are too small should not be scaled.
	if ( originalDimensions.width < SQUARE_WIDTH || originalDimensions.height < SQUARE_HEIGHT ) {
		return {
			width: originalDimensions.width,
			height: originalDimensions.height,
		};
	}

	/*
	 * If the image should be rendered as a square, and its original dimensions were also square,
	 * just use the squareWidth and squareHeight.
	 * We don't have to fear that the resulting image will be warped.
	 */
	if ( imageMode === "square" ) {
		if ( originalDimensions.width === originalDimensions.height ) {
			return {
				width: SQUARE_WIDTH,
				height: SQUARE_HEIGHT,
			};
		}

		/*
		 * If the image should be rendered as a square, but originally wasn't square,
		 * crop the longest side. This way, the image won't be warped.
		 */
		const imageRatios = getImageRatios( originalDimensions, imageMode );

		return getImageDimensionsForTwitterImage( originalDimensions, imageRatios );
	}

	/*
	 * If the image should be rendered as a landscape, crop the longest side,
	 * to reach the required size ratio. This way, the image won't be warped.
	 */
	if ( imageMode === "landscape" ) {
		const imageRatios = getImageRatios( originalDimensions, imageMode );

		return getImageDimensionsForTwitterImage( originalDimensions, imageRatios );
	}
}

/**
 * Determines the properties of the Twitter image.
 *
 * @param {string} src The source of the image.
 * @param {string} cardType The card type in which an image should be displayed.
 *
 * @returns {Promise} The promise of the imageProperties.
 */
export function determineTwitterImageProperties( src, cardType ) {
	return getOriginalImageDimensions( src ).then( ( originalDimensions ) => {
		// Determine what image mode should be used based on the card type.
		const imageMode = determineTwitterImageMode( cardType );

		// Calculate the image dimensions for the specific image.
		const TwitterImageDimensions = calculateTwitterImageDimensions( originalDimensions, imageMode );

		return {
			mode: imageMode,
			height: TwitterImageDimensions.height,
			width: TwitterImageDimensions.width,
		};
	} );
}
