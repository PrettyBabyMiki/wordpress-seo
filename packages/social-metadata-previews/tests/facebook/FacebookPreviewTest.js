/* External dependencies */
import React from "react";
import renderer from "react-test-renderer";

/* Internal dependencies */
import delayComponentSnapshot from "../testHelpers/delayComponentSnapshot";
import FacebookPreview from "../../src/facebook/FacebookPreview";

import * as determineImageProperties from "../../src/helpers/determineImageProperties";
determineImageProperties.determineImageProperties = jest.fn();

describe( "FacebookPreview Component", () => {
	it( "calls determineImageProperties with the correct image URL", () => {
		const imageUrl = "https://yoast.com/app/uploads/2015/09/Author_Joost_x2.png";

		determineImageProperties.determineImageProperties.mockReturnValueOnce( Promise.resolve( {
			mode: "portrait",
			height: 600,
			width: 300,
		} ) );

		renderer.create(
			<FacebookPreview image={ imageUrl } siteName={ "yoast.com" } title={ "yoast test" } />
		);

		expect( determineImageProperties.determineImageProperties ).toBeCalledWith( imageUrl, "Facebook" );
	} );

	it( "matches the snapshot in the loading state", () => {
		determineImageProperties.determineImageProperties.mockReturnValueOnce(
			// Make sure the promise is never resolved to keep the loading state.
			new Promise( () => {} )
		);
		const component = renderer.create(
			<FacebookPreview image="https://yoast.com/app/uploads/2015/09/Author_Joost_x2.png" siteName={ "yoast.com" } title={ "yoast test" }  />
		);
		const tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );

	it( "matches the snapshot for a portrait image", () => {
		determineImageProperties.determineImageProperties.mockReturnValueOnce( Promise.resolve( {
			mode: "portrait",
			height: 600,
			width: 300,
		} ) );
		const component = renderer.create(
			<FacebookPreview image="https://yoast.com/app/uploads/2015/09/Author_Joost_x2.png" siteName={ "yoast.com" } title={ "yoast test" }  />
		);

		// Wait for the determineImageProperties promise to resolve.
		return delayComponentSnapshot( component );
	} );

	it( "matches the snapshot for a landscape image", () => {
		determineImageProperties.determineImageProperties.mockReturnValueOnce( Promise.resolve( {
			mode: "landscape",
			height: 300,
			width: 600,
		} ) );
		const component = renderer.create(
			<FacebookPreview
				image="https://yoast.com/app/uploads/2015/06/How_to_choose_keywords_FI.png"
				siteName={ "yoast.com" }
				title={ "yoast test" }
			/>
		);

		// Wait for the determineImageProperties promise to resolve.
		return delayComponentSnapshot( component );
	} );

	it( "matches the snapshot for a square image", () => {
		determineImageProperties.determineImageProperties.mockReturnValueOnce( Promise.resolve( {
			mode: "square",
			height: 300,
			width: 300,
		} ) );
		const component = renderer.create(
			<FacebookPreview
				image="https://yoast.com/app/uploads/2018/09/avatar_user_1_1537774226.png"
				siteName={ "yoast.com" }
				title={ "yoast test" }
			/>
		);

		// Wait for the determineImageProperties promise to resolve.
		return delayComponentSnapshot( component );
	} );

	it( "matches the snapshot for a too small image", () => {
		determineImageProperties.determineImageProperties.mockReturnValue( Promise.resolve( {
			mode: "square",
			height: 100,
			width: 100,
		} ) );
		const component = renderer.create(
			<FacebookPreview image="https://yoast.com/app/uploads/2018/11/Logo_TYPO3-250x105.png" siteName={ "yoast.com" } title={ "yoast test" } />
		);

		// Wait for the determineImageProperties promise to resolve.
		return delayComponentSnapshot( component );
	} );

	it( "matches the snapshot for a faulty image", () => {
		determineImageProperties.determineImageProperties.mockReturnValueOnce( Promise.reject() );

		const component = renderer.create(
			<FacebookPreview image="thisisnoimage" siteName={ "yoast.com" } title={ "yoast test" }  />
		);

		// Wait for the determineImageProperties promise to resolve.
		return delayComponentSnapshot( component );
	} );
} );

describe( "FacebookPreview", () => {
	it( "matches the snapshot for a landscape image", () => {
		const component = renderer.create(
			<FacebookPreview
				siteName="yoast.com"
				title="YoastCon Workshops"
				description="Description to go along with a landscape image."
				src="https://yoast.com/app/uploads/2015/06/How_to_choose_keywords_FI.png"
			/>
		);

		const tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );
	it( "matches the snapshot for a portrait image", () => {
		const component = renderer.create(
			<FacebookPreview
				siteName="yoast.com"
				title="YoastCon Workshops"
				description="Description to go along with a portrait image."
				src="https://yoast.com/app/uploads/2015/09/Author_Joost_x2.png"
			/>
		);

		const tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );
	it( "matches the snapshot for a square image", () => {
		const component = renderer.create(
			<FacebookPreview
				siteName="yoast.com"
				title="YoastCon Workshops"
				description="Description to go along with a square image."
				src="https://yoast.com/app/uploads/2018/09/avatar_user_1_1537774226.png"
			/>
		);

		const tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );
	it( "matches the snapshot for a too small image", () => {
		const component = renderer.create(
			<FacebookPreview
				siteName="yoast.com"
				title="YoastCon Workshops"
				description="Description to go along with too small an image."
				src="https://yoast.com/app/uploads/2018/11/Logo_TYPO3-250x105.png"
			/>
		);

		const tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );
	it( "matches the snapshot for a faulty image", () => {
		const component = renderer.create(
			<FacebookPreview
				siteName="yoast.com"
				title="YoastCon Workshops"
				description="Description to go along with a faulty image."
				src="thisisnoimage"
			/>
		);

		const tree = component.toJSON();
		expect( tree ).toMatchSnapshot();
	} );
} );
