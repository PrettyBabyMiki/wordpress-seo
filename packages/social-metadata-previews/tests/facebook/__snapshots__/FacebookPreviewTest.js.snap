// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FacebookPreview matches the snapshot for a faulty image 1`] = `
Array [
  .c0 {
  width: 500px;
  height: 261px;
  background-color: #ddd;
}

<div
    className="c0"
  />,
  <span
    className="screen-reader-text"
  >
    yoast.com
  </span>,
  .c1 {
  color: #606770;
  font-size: 12px;
  line-height: 11px;
  text-transform: uppercase;
  overflow: hidden;
}

.c0 {
  color: #606770;
  font-size: 12px;
  line-height: 11px;
  overflow: hidden;
  text-overflow: ellipsis;
  text-transform: uppercase;
  white-space: nowrap;
}

<p
    aria-hidden="true"
    className="c0"
  >
    <span
      className="c1"
    >
      yoast.com
    </span>
  </p>,
  .c0 {
  display: block;
  color: #1d2129;
  font-weight: 600;
  overflow: hidden;
  font-size: 16px;
  line-height: 20px;
  margin: 5px 0 0;
  max-height: 100px;
  word-wrap: break-word;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  white-space: normal;
  cursor: pointer;
}

<span
    className="c0"
  >
    YoastCon Workshops
  </span>,
  .c0 {
  color: #606770;
  font-size: 14px;
  line-height: 20px;
  word-break: break-word;
  max-height: 80px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  margin-top: 3px;
  width: 100%;
}

<p
    className="c0"
  >
    Description to go along with a faulty image.
  </p>,
]
`;

exports[`FacebookPreview matches the snapshot for a landscape image 1`] = `
Array [
  .c0 {
  width: 500px;
  height: 261px;
  background-color: #ddd;
}

<div
    className="c0"
  />,
  <span
    className="screen-reader-text"
  >
    yoast.com
  </span>,
  .c1 {
  color: #606770;
  font-size: 12px;
  line-height: 11px;
  text-transform: uppercase;
  overflow: hidden;
}

.c0 {
  color: #606770;
  font-size: 12px;
  line-height: 11px;
  overflow: hidden;
  text-overflow: ellipsis;
  text-transform: uppercase;
  white-space: nowrap;
}

<p
    aria-hidden="true"
    className="c0"
  >
    <span
      className="c1"
    >
      yoast.com
    </span>
  </p>,
  .c0 {
  display: block;
  color: #1d2129;
  font-weight: 600;
  overflow: hidden;
  font-size: 16px;
  line-height: 20px;
  margin: 5px 0 0;
  max-height: 100px;
  word-wrap: break-word;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  white-space: normal;
  cursor: pointer;
}

<span
    className="c0"
  >
    YoastCon Workshops
  </span>,
  .c0 {
  color: #606770;
  font-size: 14px;
  line-height: 20px;
  word-break: break-word;
  max-height: 80px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  margin-top: 3px;
  width: 100%;
}

<p
    className="c0"
  >
    Description to go along with a landscape image.
  </p>,
]
`;

exports[`FacebookPreview matches the snapshot for a portrait image 1`] = `
Array [
  .c0 {
  width: 500px;
  height: 261px;
  background-color: #ddd;
}

<div
    className="c0"
  />,
  <span
    className="screen-reader-text"
  >
    yoast.com
  </span>,
  .c1 {
  color: #606770;
  font-size: 12px;
  line-height: 11px;
  text-transform: uppercase;
  overflow: hidden;
}

.c0 {
  color: #606770;
  font-size: 12px;
  line-height: 11px;
  overflow: hidden;
  text-overflow: ellipsis;
  text-transform: uppercase;
  white-space: nowrap;
}

<p
    aria-hidden="true"
    className="c0"
  >
    <span
      className="c1"
    >
      yoast.com
    </span>
  </p>,
  .c0 {
  display: block;
  color: #1d2129;
  font-weight: 600;
  overflow: hidden;
  font-size: 16px;
  line-height: 20px;
  margin: 5px 0 0;
  max-height: 100px;
  word-wrap: break-word;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  white-space: normal;
  cursor: pointer;
}

<span
    className="c0"
  >
    YoastCon Workshops
  </span>,
  .c0 {
  color: #606770;
  font-size: 14px;
  line-height: 20px;
  word-break: break-word;
  max-height: 80px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  margin-top: 3px;
  width: 100%;
}

<p
    className="c0"
  >
    Description to go along with a portrait image.
  </p>,
]
`;

exports[`FacebookPreview matches the snapshot for a square image 1`] = `
Array [
  .c0 {
  width: 500px;
  height: 261px;
  background-color: #ddd;
}

<div
    className="c0"
  />,
  <span
    className="screen-reader-text"
  >
    yoast.com
  </span>,
  .c1 {
  color: #606770;
  font-size: 12px;
  line-height: 11px;
  text-transform: uppercase;
  overflow: hidden;
}

.c0 {
  color: #606770;
  font-size: 12px;
  line-height: 11px;
  overflow: hidden;
  text-overflow: ellipsis;
  text-transform: uppercase;
  white-space: nowrap;
}

<p
    aria-hidden="true"
    className="c0"
  >
    <span
      className="c1"
    >
      yoast.com
    </span>
  </p>,
  .c0 {
  display: block;
  color: #1d2129;
  font-weight: 600;
  overflow: hidden;
  font-size: 16px;
  line-height: 20px;
  margin: 5px 0 0;
  max-height: 100px;
  word-wrap: break-word;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  white-space: normal;
  cursor: pointer;
}

<span
    className="c0"
  >
    YoastCon Workshops
  </span>,
  .c0 {
  color: #606770;
  font-size: 14px;
  line-height: 20px;
  word-break: break-word;
  max-height: 80px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  margin-top: 3px;
  width: 100%;
}

<p
    className="c0"
  >
    Description to go along with a square image.
  </p>,
]
`;

exports[`FacebookPreview matches the snapshot for a too small image 1`] = `
Array [
  .c0 {
  width: 500px;
  height: 261px;
  background-color: #ddd;
}

<div
    className="c0"
  />,
  <span
    className="screen-reader-text"
  >
    yoast.com
  </span>,
  .c1 {
  color: #606770;
  font-size: 12px;
  line-height: 11px;
  text-transform: uppercase;
  overflow: hidden;
}

.c0 {
  color: #606770;
  font-size: 12px;
  line-height: 11px;
  overflow: hidden;
  text-overflow: ellipsis;
  text-transform: uppercase;
  white-space: nowrap;
}

<p
    aria-hidden="true"
    className="c0"
  >
    <span
      className="c1"
    >
      yoast.com
    </span>
  </p>,
  .c0 {
  display: block;
  color: #1d2129;
  font-weight: 600;
  overflow: hidden;
  font-size: 16px;
  line-height: 20px;
  margin: 5px 0 0;
  max-height: 100px;
  word-wrap: break-word;
  -webkit-letter-spacing: normal;
  -moz-letter-spacing: normal;
  -ms-letter-spacing: normal;
  letter-spacing: normal;
  white-space: normal;
  cursor: pointer;
}

<span
    className="c0"
  >
    YoastCon Workshops
  </span>,
  .c0 {
  color: #606770;
  font-size: 14px;
  line-height: 20px;
  word-break: break-word;
  max-height: 80px;
  overflow: hidden;
  text-overflow: ellipsis;
  white-space: nowrap;
  margin-top: 3px;
  width: 100%;
}

<p
    className="c0"
  >
    Description to go along with too small an image.
  </p>,
]
`;
